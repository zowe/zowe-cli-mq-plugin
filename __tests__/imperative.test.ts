/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

describe("imperative config", () => {

    // Will fail if imperative config object is changed. This is a sanity/protection check to ensure that any
    // changes to the configuration document are intended.
    // Removed snapshot due to pluginHealthCheck path varies from machine to machine.
    it("config should match expected values", () => {
        const config = require("../src/imperative");
        expect(config.name).toBe("mq");
        expect(config.pluginHealthCheck).toContain("healthCheck.Handler");
        expect(config.productDisplayName).toBe("CLI Plug-in for IBM MQ for z/OS.");
        expect(config.rootCommandDescription).toContain("Interact with IBM MQ for z/OS.");
    });

    // Will fail if imperative config object is changed. This is a sanity/protection check to ensure that any
    // changes to the configuration document are intended (and the snapshot must be updated).
    it("should match the snapshot", () => {
        const config = require("../src/imperative");
        delete config.pluginHealthCheck; // this path changes depending on your system and can't be snapshotted.
        expect(config).toMatchSnapshot();
    });
});
