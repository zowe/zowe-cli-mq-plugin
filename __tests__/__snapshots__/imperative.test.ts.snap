// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imperative config should match the snapshot 1`] = `
Object {
  "commandModuleGlobs": Array [
    "**/cli/*/*.definition!(.d).*s",
  ],
  "name": "mq",
  "productDisplayName": "CLI Plug-in for IBM MQ for z/OS.",
  "profiles": Array [
    Object {
      "createProfileExamples": Array [
        Object {
          "description": "Create an MQ profile named 'mqprofile' to connect to MQ at host zos123 and port 1234",
          "options": "mq --host mq123 --port 1234 --user ibmuser --password myp4ss",
        },
      ],
      "schema": Object {
        "description": "An MQREST profile is required to issue commands in the MQ command group that interacts with MQSC. The mq profile contains your host, port, user name, and password for the IBM MQ System Console interface",
        "properties": Object {
          "host": Object {
            "optionDefinition": Object {
              "aliases": Array [
                "H",
              ],
              "description": "The host name used to access the IBM MQ REST API. This might be the host name of the IBM MQ mqweb server, or the Zowe API Mediation Layer..",
              "group": "MQ Connection Options",
              "name": "host",
              "type": "string",
            },
            "type": "string",
          },
          "password": Object {
            "optionDefinition": Object {
              "aliases": Array [
                "pass",
                "pw",
              ],
              "description": "The mainframe (MQ) password, which can be the same as your TSO password.",
              "group": "MQ Connection Options",
              "name": "password",
              "type": "string",
            },
            "secure": true,
            "type": "string",
          },
          "port": Object {
            "optionDefinition": Object {
              "aliases": Array [
                "P",
              ],
              "description": "The port number used to access the IBM MQ REST API. This might be the port number of the IBM MQ mqweb server, or the Zowe API Mediation Layer.",
              "group": "MQ Connection Options",
              "name": "port",
              "type": "number",
            },
            "type": "number",
          },
          "protocol": Object {
            "optionDefinition": Object {
              "allowableValues": Object {
                "caseSensitive": false,
                "values": Array [
                  "http",
                  "https",
                ],
              },
              "defaultValue": "https",
              "description": "Specifies the MQ protocol (http or https).",
              "group": "MQ Connection Options",
              "name": "protocol",
              "type": "string",
            },
            "type": "string",
          },
          "rejectUnauthorized": Object {
            "optionDefinition": Object {
              "aliases": Array [
                "ru",
              ],
              "defaultValue": false,
              "description": "Reject self-signed certificates.",
              "group": "MQ Connection Options",
              "name": "reject-unauthorized",
              "type": "boolean",
            },
            "type": "boolean",
          },
          "user": Object {
            "optionDefinition": Object {
              "aliases": Array [
                "u",
              ],
              "description": "The mainframe (MQ) user name, which can be the same as your TSO login.",
              "group": "MQ Connection Options",
              "name": "user",
              "type": "string",
            },
            "secure": true,
            "type": "string",
          },
        },
        "required": Array [],
        "title": "CLI profile for MQ",
        "type": "object",
      },
      "type": "mq",
    },
  ],
  "rootCommandDescription": "Interact with IBM MQ for z/OS.",
}
`;
